nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/ConsumeTwitter.java

```diff
@@ -338,15 +343,43 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
         session.getProvenanceReporter().receive(flowFile, transitUri);
     }
 
+    @OnPrimaryNodeStateChange
+    public void onPrimaryNodeStateChange(final PrimaryNodeState newState) {
+        if (newState == PrimaryNodeState.PRIMARY_NODE_REVOKED) {
+            stopTweetStreamService(true);
+        }
+    }
+
     @OnStopped
     public void onStopped() {
-        if (tweetStreamService != null) {
-            tweetStreamService.stop();
-        }
-        tweetStreamService = null;
+        stopTweetStreamService(false);
         emptyQueue();
     }
 
+    private synchronized void startTweetStreamService(final ProcessContext context) {
+        if (isInitialized.compareAndSet(false, true)) {
+            tweetStreamService = new TweetStreamService(context, messageQueue, getLogger());
+            tweetStreamService.start();
+        }
+
+    }
+
+    private synchronized void stopTweetStreamService(final boolean printMessageQueueWarning) {
```

Comment:
The `synchronized` modifier does not seem necessary given the use of `AtomicBoolean` for tracking status.
In addition, the `printMessageQueueWarning` seems unnecessary, it would be better to log a more generic warning in all cases.